client/src/main/scala/drt/client/services/handlers/InitialCrunchStateHandler.scala

```diff
@@ -0,0 +1,82 @@
+package drt.client.services.handlers
+
+import diode._
+import diode.Implicits.runAfterImpl
+import diode.data._
+import drt.client.actions.Actions._
+import drt.client.logger._
+import drt.client.services._
+import drt.shared.CrunchApi._
+import org.scalajs.dom
+import upickle.default.read
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
+
+class InitialCrunchStateHandler[M](getCurrentViewMode: () => ViewMode,
+                                   modelRW: ModelRW[M, (Pot[CrunchState], MillisSinceEpoch)]) extends LoggingActionHandler(modelRW) {
+  val crunchUpdatesRequestFrequency: FiniteDuration = 2 seconds
+
+  def startMillisFromView: MillisSinceEpoch = getCurrentViewMode().dayStart.millisSinceEpoch
+
+  def viewHasChanged(viewMode: ViewMode): Boolean = viewMode.dayStart.millisSinceEpoch != startMillisFromView
+
+  def handle: PartialFunction[Any, ActionResult[M]] = {
+    case GetInitialCrunchState(viewMode) =>
+      val startMillis = startMillisFromView
+      val endMillis = startMillis + (1000 * 60 * 60 * 36)
```

Comment:
perhaps put this into a thirtySixHours val

Reply:
Done